peer/peer.go

```diff
@@ -0,0 +1,2027 @@
+		filterClearMsgListeners: make(map[string]func(*Peer, *wire.MsgFilterClear)),
+		filterLoadMsgListeners:  make(map[string]func(*Peer, *wire.MsgFilterLoad)),
+		versionMsgListeners:     make(map[string]func(*Peer, *wire.MsgVersion)),
+		rejectMsgListeners:      make(map[string]func(*Peer, *wire.MsgReject)),
+	}
+	return &p
+}
+
+// NewInboundPeer returns a new inbound bitcoin peer for the provided server
+// and connection.  Use Start to begin processing incoming and outgoing
+// messages.
+func NewInboundPeer(newestSha ShaFunc, nonce uint64, btcnet wire.BitcoinNet, services wire.ServiceFlag, addrManager *addrmgr.AddrManager, cfg *Config, conn net.Conn) *Peer {
+	p := newPeerBase(newestSha, nonce, btcnet, addrManager, cfg, services, true)
+	p.conn = conn
+	p.addr = conn.RemoteAddr().String()
+	p.timeConnected = time.Now()
+	atomic.AddInt32(&p.connected, 1)
+	return p
+}
+
+// NewOutboundPeer returns a new outbound bitcoin peer for the provided server
+// and address and connects to it asynchronously. If the connection is
+// successful then the peer will also be started.
+func NewOutboundPeer(newestSha ShaFunc, nonce uint64, btcnet wire.BitcoinNet, services wire.ServiceFlag, addrManager *addrmgr.AddrManager, cfg *Config, conn net.Conn, addr string) *Peer {
+	p := newPeerBase(newestSha, nonce, btcnet, addrManager, cfg, services, false)
+	p.addr = addr
+	p.conn = conn
+
+	// Setup p.na with a temporary address that we are connecting to with
+	// faked up service flags.  We will replace this with the real one after
+	// version negotiation is successful.  The only failure case here would
+	// be if the string was incomplete for connection so can't be split
+	// into address and port, and thus this would be invalid anyway.  In
+	// which case we return nil to be handled by the caller.  This must be
+	// done before we fork off the goroutine because as soon as this
+	// function returns the peer must have a valid netaddress.
+	host, portStr, err := net.SplitHostPort(addr)
+	if err != nil {
+		p.LogError("Tried to create a new outbound peer with invalid "+
+			"address %s: %v", addr, err)
+		return nil
+	}
+
+	port, err := strconv.ParseUint(portStr, 10, 16)
+	if err != nil {
+		p.LogError("Tried to create a new outbound peer with invalid "+
+			"port %s: %v", portStr, err)
+		return nil
+	}
+
+	p.na, err = p.addrManager.HostToNetAddress(host, uint16(port), 0)
+	if err != nil {
+		p.LogError("Can not turn host %s into netaddress: %v",
+			host, err)
+		return nil
+	}
+
+	log.Debugf("Attempting to connect to %s", p.Addr())
+	p.timeConnected = time.Now()
+	p.addrManager.Attempt(p.na)
+
+	// Connection was successful so log it and start peer.
+	log.Debugf("Connected to %s", p.conn.RemoteAddr())
+	atomic.AddInt32(&p.connected, 1)
+	p.Start()
+
+	return p
+}
+
+// LogError makes sure that we only log errors loudly on user peers.
+func (p *Peer) LogError(fmt string, args ...interface{}) {
```

Comment:
There is no point in having this function once the persistent stuff was removed.  That said, it was intentionally doing different logging based upon whether it was a persistent peer, so you'll need to carefully examine why it was doing that and retrain (or otherwise) improve the behavior that made it necessary to begin with.